### ISSUE TITLE
Cold Blood + Mutilate.

### ISSUE THREAD

**Issue by lineagedr** [TrinityCore/TrinityCore#20421]
**Description:**
Mutilate doesn't crit for the two attacks it creates after using Cold Blood.

**Steps to reproduce the problem:**

1. Cast Cold Blood.
2. Cast Mutilate.

It should crit for the main hand and off hand attacks.

Currently it only crits for main hand attack. and off attack doesn't crit.

**Branch(es):**
3.3.5a.

**TC rev. hash/commit:** 
https://github.com/TrinityCore/TrinityCore/commit/cb9b9f7247c57ee0eed42b063b827799166ce20a

**TDB version:** 
TDB 335.63.

**Operating system:**
Win64.

**Comment by GeForceBG** [comment]
confirm

**Comment by tkrokli** [comment]
@GeForceBG : Please update your comment with your own core rev. hash when confirming. Confirmations without core rev. hash is practically useless (for tracking purposes).

**Comment by lineagedr** [comment]
Any news on this?

**Comment by tkrokli** [comment]
No, I don't see anything new happening here.
Do you have any new or additional information to add to the issue?

**Comment by lineagedr** [comment]
@tkrokli Nothing new, maybe this old commit could be of some help https://github.com/kelno/windrunner/commit/cf1a92e7b84e95d89add4b2b178ae091b5a002f3

**Comment by tkrokli** [comment]
```sql
UPDATE spell_affect SET SpellFamilyMask = SpellFamilyMask | 0x200000000 WHERE entry IN (14177);
```
That repository looks like a MaNGOS or CMaNGOS clone, (I could be wrong, though). I am trying to track what happened to this kind of data during the transition from MaNGOS to today's TrinityCore source, but it's a bit difficult. The only code I could find on short notice related to `SpellFamilyMask` was in some old SQLs from 2.4.3 (TBC) and in the table `spell_proc`.

You might want to search through the existing code instead and check how it is handled:
https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/AI/PlayerAI/PlayerAI.cpp#L201-L205
https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/AI/PlayerAI/PlayerAI.cpp#L465-L466
https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/AI/PlayerAI/PlayerAI.cpp#L946-L948

The MaNGOS SQL looks more like a `spell_linked_spell` hack if you ask me, but I don't know enough about it.

**Comment by lineagedr** [comment]
@tkrokli I think cold blood is removed after mutilates main hand attack resulting in off-hand attack never criting.

**Comment by tkrokli** [comment]
Fair enough. I am no expert on the spells used by rogues (or any spell mechanics for that matter).
I hope someone who knows more will reply and give some useful feedback.

**Comment by ariel-** [comment]
Some sort of info between spell effects is needed here; Mutilate is a dummy spell triggering 2 other spells, one for each hand.

Seal Fate has a very similar issue currently: it will only proc extra combo point whenever main hand crits, disregarding hit result of off-hand completely, and needs handling for case when both hands crit should proc only once.

Maybe hit result for triggered spell must be calculated before casting? (ie while processing triggering spell)

**Comment by killerwife** [comment]
A crit from Mutilate should only give ONE additional combo point, not two, when used with Seal Fate.

This comment explains how it should work:
https://github.com/FeenixServerProject/Archangel_2.4.3_Bugtracker/issues/2382#issuecomment-13767027
